org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/validation/ProjectAwareUniqueClassNameValidator.xtend

```diff
@@ -174,15 +174,17 @@ class ProjectAwareUniqueClassNameValidator extends UniqueClassNameValidator {
 
 	def protected isDerived(IResource resource) {
 		try {
-			if (derivedResourceMarkers.findDerivedResourceMarkers(resource).length >= 1) {
+			if (derivedResourceMarkers !== null && derivedResourceMarkers.findDerivedResourceMarkers(resource).length >= 1) {
```

Comment:
derivedResourceMarkers should never be null. It was done like this because of the mocked stuff above?

Reply:
Yes